src/ripple/rpc/handlers/LedgerHandler.cpp

```diff
@@ -134,27 +134,37 @@ doLedgerGrpc(RPC::GRPCContext<org::xrpl::rpc::v1::GetLedgerRequest>& context)
 
     if (request.transactions())
     {
-        for (auto& i : ledger->txs)
+        try
         {
-            assert(i.first);
-            if (request.expand())
+            for (auto& i : ledger->txs)
             {
-                auto txn =
-                    response.mutable_transactions_list()->add_transactions();
-                Serializer sTxn = i.first->getSerializer();
-                txn->set_transaction_blob(sTxn.data(), sTxn.getLength());
-                if (i.second)
+                assert(i.first);
+                if (request.expand())
                 {
-                    Serializer sMeta = i.second->getSerializer();
-                    txn->set_metadata_blob(sMeta.data(), sMeta.getLength());
+                    auto txn = response.mutable_transactions_list()
+                                   ->add_transactions();
+                    Serializer sTxn = i.first->getSerializer();
+                    txn->set_transaction_blob(sTxn.data(), sTxn.getLength());
+                    if (i.second)
+                    {
+                        Serializer sMeta = i.second->getSerializer();
+                        txn->set_metadata_blob(sMeta.data(), sMeta.getLength());
+                    }
+                }
+                else
+                {
+                    auto const& hash = i.first->getTransactionID();
+                    response.mutable_hashes_list()->add_hashes(
+                        hash.data(), hash.size());
                 }
             }
-            else
-            {
-                auto const& hash = i.first->getTransactionID();
-                response.mutable_hashes_list()->add_hashes(
-                    hash.data(), hash.size());
-            }
+        }
+        catch (std::exception const& e)
+        {
+            JLOG(context.j.error())
+                << __func__ << " - Error deserializing transaction in ledger "
+                << ledger->info().seq
+                << " . skipping transaction. you should look into this further";
```

Comment:
Since the try/catch surrounds the entire `for` loop, I don't think that what is written to the log is accurate.  It's not just one transaction that is being skipped.  It's also any transactions in the ledger that might follow that transaction.  I'm okay with that implementation if it works for clio, but maybe the message should be accurate.